### PR TITLE
Fix error when update failed 5 times

### LINKED ISSUE #6140: Error: Update failed 5 times
This happens when we try `manualUpdate`. We do not handle the error at all on the client side.

```
Client: darwin
Tutanota version: 3.118.13
User agent:
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) tutanota-desktop/3.118.13 Chrome/116.0.5845.190 Electron/26.2.2 Safari/537.36
Error
Error message: Update failed 5 times
Stacktrace:
Error: Update failed 5 times
at At.manualUpdate (/Applications/Tutanota [Desktop.app/Contents/Resources/app.asar/desktop/DesktopMain.js:1:59066](http://desktop.app/Contents/Resources/app.asar/desktop/DesktopMain.js:1:59066))
at vT.manualUpdate (/Applications/Tutanota [Desktop.app/Contents/Resources/app.asar/desktop/DesktopMain.js:2:288426](http://desktop.app/Contents/Resources/app.asar/desktop/DesktopMain.js:2:288426))
at MT.dispatch (/Applications/Tutanota [Desktop.app/Contents/Resources/app.asar/desktop/DesktopMain.js:2:294892](http://desktop.app/Contents/Resources/app.asar/desktop/DesktopMain.js:2:294892))
at BT.dispatch (/Applications/Tutanota [Desktop.app/Contents/Resources/app.asar/desktop/DesktopMain.js:2:297624](http://desktop.app/Contents/Resources/app.asar/desktop/DesktopMain.js:2:297624))
at ipc (/Applications/Tutanota [Desktop.app/Contents/Resources/app.asar/desktop/DesktopMain.js:2:287919](http://desktop.app/Contents/Resources/app.asar/desktop/DesktopMain.js:2:287919))
at exports.MessageDispatcher.handleMessage (/Applications/Tutanota [Desktop.app/Contents/Resources/app.asar/desktop/NodeWorkerTransport-7c28b2d6.js:1:55142](http://desktop.app/Contents/Resources/app.asar/desktop/NodeWorkerTransport-7c28b2d6.js:1:55142))
at /Applications/Tutanota [Desktop.app/Contents/Resources/app.asar/desktop/NodeWorkerTransport-7c28b2d6.js:1:54251](http://desktop.app/Contents/Resources/app.asar/desktop/NodeWorkerTransport-7c28b2d6.js:1:54251)
at /Applications/Tutanota [Desktop.app/Contents/Resources/app.asar/desktop/DesktopMain.js:2:280396](http://desktop.app/Contents/Resources/app.asar/desktop/DesktopMain.js:2:280396)
at WebContents.<anonymous> (node:electron/js2c/browser_init:2:89468)
at WebContents.emit (node:events:513:28)
```

## Test Notes
- [ ] The app still searching for updates when startup
- [ ] When a update is available the app can download and install it
- [ ] The user can check for updates
- [ ] If the update fails more than 4 times, every time the user checks for a update it's throttled by (500 * tries)ms
- [ ] If the update fails more than 4 times, it will show a notification

### PR DESCRIPTION
**Pull request by murilopereirame**
When the update fails 5 times, the ElectronUpdater has its listeners removed, the user receives a notification informing that the software will try to update later and if the user tries to update again, they receive an error message.

Since there's no more listeners, and we don't want a massive check for updates on our servers, the user doesn't receive the error anymore, just the checkUpdate promise as is resolved as false.

fix #6140

### CODE REVIEW

**Review by charlag — COMMENTED** [review]
Should we re-discuss this? I think we should stop polling after 5 times but I don't think we should stop trying after 5 times if the user does a manual update check.

**Comment by murilopereirame** [comment]
> Should we re-discuss this? I think we should stop polling after 5 times but I don't think we should stop trying after 5 times if the user does a manual update check.

Maybe, I think we need to prevent the user of spamming the check for updates button, at least put a timeout after the 5th try? And don't remove the listeners after 5 errors

**Comment by charlag** [comment]
I would just throttle it a bit

File: src/desktop/ElectronUpdater.ts [hunk 1]
```diff
@@ -215,6 +214,17 @@ export class ElectronUpdater {
 		}
 	}
 
+	private async checkUpdateThrottleTime() {
+		// After 5 unsuccessful attempts to update, we start throttling the user
+		// to avoid too many calls to our servers
+		if (this.errorCount >= 5) {
+			// Half of a second * errorCount
+			const throttleTime = 500 * this.errorCount
+			this.logger.debug(`Throttling user ${throttleTime} - Try N ${this.errorCount}`)
```
Comment (charlag) [inline src/desktop/ElectronUpdater.ts:223]: I would just use `debounceStart` with some fixed time but I think yours should also be good.

I would change the message to `Auto Update: throttling manual update attempt # ${this.errorCount} by ${throttleTime}`

**Review by charlag — APPROVED** [review]

**Pull request merged**